magellan-library/src/main/java/com/wealthfront/magellan/lifecycle/LifecycleTransitionExtensions.kt

```diff
@@ -0,0 +1,77 @@
+package com.wealthfront.magellan.lifecycle
+
+import android.content.Context
+import com.wealthfront.magellan.lifecycle.LifecycleState.Created
+import com.wealthfront.magellan.lifecycle.LifecycleState.Destroyed
+import com.wealthfront.magellan.lifecycle.LifecycleState.Resumed
+import com.wealthfront.magellan.lifecycle.LifecycleState.Shown
+import com.wealthfront.magellan.lifecycle.LifecycleStateDirection.BACKWARDS
+import com.wealthfront.magellan.lifecycle.LifecycleStateDirection.FORWARD
+import com.wealthfront.magellan.lifecycle.LifecycleStateDirection.NO_MOVEMENT
+
+public fun <T> T.transitionToState(newState: LifecycleState) where T : LifecycleAware, T : LifecycleOwner {
+  transition(this.currentState, newState)
+}
+
+public fun LifecycleAware.transition(oldState: LifecycleState, newState: LifecycleState) {
```

Comment:
I must be thinking about this backwards. Aren't `LifecycleOwner`s the ones with state, shouldn't they be transitioning rather than `LifecycleAware`s? 

Is this simulating how "parents" drive their "child"s lifecycle?

Reply:
This is actually the function parents (well, technically `LifecycleRegistry`) call to move their children through their lifecycles, not even just simulating it.

I'm open to renaming things, but currently `LifecycleAware` is the interface for receiving events, therefore that that's the one we "drive" here by providing lifecycle events.